### PR TITLE
test: implement miner sector event extractor tests

### PR DESCRIPTION
**Pull request by frrist**

### CODE REVIEW

File: tasks/actorstate/miner/sector_events.go [hunk 1]
```diff
@@ -25,27 +26,27 @@ func (SectorEventsExtractor) Extract(ctx context.Context, a actorstate.ActorInfo
...
+		sectors, err := ec.CurrentState().LoadSectors(nil)
 		if err != nil {
 			return nil, fmt.Errorf("loading miner sectors: %w", err)
 		}
 
 		sectorChanges = miner.MakeSectorChanges()
-		for i, sector := range sectors {
-			sectorChanges.Added[i] = *sector
+		for _, sector := range sectors {
+			sectorChanges.Added = append(sectorChanges.Added, *sector)
```
Comment (frrist) [inline tasks/actorstate/miner/sector_events.go:45]: Fixes a panic as we need to append to `Added`. This is a rare case when processing the genesis block.

File: tasks/actorstate/miner/sector_events.go [hunk 2]
```diff
@@ -54,37 +55,42 @@ func (SectorEventsExtractor) Extract(ctx context.Context, a actorstate.ActorInfo
...
 
-		sectorChanges, err = node.DiffSectors(ctx, a.Address, a.Current, a.Executed, ec.PrevState, ec.CurrState)
+		sectorChanges, err = node.DiffSectors(ctx, a.Address, a.Current, a.Executed, ec.ParentState(), ec.CurrentState())
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	sectorEventModel, err := extractSectorEvents(ctx, a, ec, sectorChanges, preCommitChanges)
+	dlDiff, err := miner.DiffDeadlines(ec.ParentState(), ec.CurrentState())
```
Comment (frrist) [inline tasks/actorstate/miner/sector_events.go:69]: moved to top-level method to improve testability of the different ExtractSector* methods.

**Review by kasteph — APPROVED** [review]

File: tasks/actorstate/miner/sector_events_test.go [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+		require.Equal(t, int64(ts.Height()), e.Height)
+		require.Equal(t, addr.String(), e.MinerID)
+		require.Equal(t, ts.ParentState().String(), e.StateRoot)
+		require.EqualValues(t, i, e.SectorID)
+	}
+
+	recovering := events[minermodel.SectorRecovering]
+	sort.Slice(recovering, func(i, j int) bool {
+		return recovering[i].SectorID < recovering[j].SectorID
+	})
```
Comment (placer14) [inline tasks/actorstate/miner/sector_events_test.go:109]:
```suggestion
	require.Equal(t, sort.Slice(recovering, func(i, j int) bool {
		return recovering[i].SectorID < recovering[j].SectorID
	}), true)
```

File: tasks/actorstate/miner/sector_events_test.go [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+		require.Equal(t, addr.String(), e.MinerID)
+		require.Equal(t, ts.ParentState().String(), e.StateRoot)
+		require.EqualValues(t, i, e.SectorID)
+	}
+
+	snapped := events[minermodel.SectorSnapped]
+	sort.Slice(snapped, func(i, j int) bool {
+		return snapped[i].SectorID < snapped[j].SectorID
+	})
+	for i, e := range events[minermodel.SectorSnapped] {
```
Comment (placer14) [inline tasks/actorstate/miner/sector_events_test.go:329]:
```suggestion
	for i, e := range snapped {
```

File: tasks/actorstate/miner/sector_events_test.go [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
```
Comment (placer14) [inline tasks/actorstate/miner/sector_events_test.go:211]:
```suggestion
	for i, e := range snapped {
```

File: tasks/actorstate/miner/sector_events_test.go [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+	}
+	for i := uint64(0); i < rm; i++ {
+		removed[i] = generateFakeSectorPreCommitOnChainInfo(i)
+	}
+
+	return &miner.PreCommitChanges{
+		Added:   added,
+		Removed: removed,
+	}
+
```
Comment (placer14) [inline tasks/actorstate/miner/sector_events_test.go:464]: nit:
```suggestion
```

File: tasks/actorstate/miner/sector_events_test.go [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+		return precommit[i].SectorID < precommit[j].SectorID
+	})
+	for i, res := range precommit {
+		require.Equal(t, addr.String(), res.MinerID)
+		require.Equal(t, int64(ts.Height()), res.Height)
+		require.Equal(t, ts.ParentState().String(), res.StateRoot)
+		require.Equal(t, minermodel.PreCommitAdded, res.Event)
+		require.EqualValues(t, fakePrecommitChanges.Added[i].Info.SectorNumber, res.SectorID)
+	}
+
```
Comment (placer14) [inline tasks/actorstate/miner/sector_events_test.go:396]: nit:
```suggestion
```

**Review by placer14 — APPROVED** [review]
Really minor nits here on sort if you think it's important to consider.

File: tasks/actorstate/miner/sector_events_test.go [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+	events := make(map[string][]*minermodel.MinerSectorEvent)
+	for _, res := range result {
+		events[res.Event] = append(events[res.Event], res)
+	}
+
+	terminated := events[minermodel.SectorTerminated]
+	sort.Slice(terminated, func(i, j int) bool {
+		return terminated[i].SectorID < terminated[j].SectorID
+	})
```
[the remark this answers is not in the document]
Comment (frrist) [reply tasks/actorstate/miner/sector_events_test.go:73]: I'm not sure I understand, could you explain more?
`sort.Slice` doesn't return anything: https://pkg.go.dev/sort#Slice

File: tasks/actorstate/miner/sector_events_test.go (hunk 5 above)
Comment (frrist) [reply tasks/actorstate/miner/sector_events_test.go:211]: Good catch, will address

**Pull request merged**